src/services/user.service.js

```
@@ -0,0 +1,72 @@
+import { User } from '../models/index.js';
+import { ApiError } from '../exceptions/api.error.js';
+
+const createNewUser = async (username) => {
+  if (!username) {
+    throw ApiError.badRequest('Username is required to create new user');
+  }
+
+  const userAlreadyExist = await findByUserName(username);
+
+  if (userAlreadyExist) {
+    throw ApiError.conflict('This username is already in use');
+  }
+
+  return User.create({ username });
+};
+
+const normalizeUser = ({ id, username }) => {
+  return { id, username };
+};
+
+const findByUserName = async (username) => {
+  if (!username) {
+    throw ApiError.badRequest('Username is required to find user');
+  }
+
+  const user = await User.findOne({
+    where: {
+      username,
+    },
+  });
+
+  if (!user) {
+    throw ApiError.notFound('User with such username is not found');
```

The function throws a `notFound` error when a user with such username is not found, but it also throws the same error if the username is not provided. It should throw a `badRequest` error if the username is missing.